src/towncrier/test/test_create.py

```diff
@@ -151,12 +186,72 @@ def test_invalid_section(self):
             "Expected filename '123.foobar.rst' to be of format", result.output
         )
 
+    @with_isolated_runner
+    def test_custom_extension(self, runner: CliRunner):
+        """Ensure we can still create fragments with custom extensions."""
+        setup_simple_project()
+        frag_path = Path("foo", "newsfragments")
+
+        result = runner.invoke(_main, ["123.feature.txt"])
+        self.assertEqual(result.exit_code, 0, result.output)
+
+        fragments = [f.name for f in frag_path.iterdir()]
+        # No '.rst' extension added.
+        self.assertEqual(fragments, ["123.feature.txt"])
+
+    @with_isolated_runner
+    def test_md_filename_extension(self, runner: CliRunner):
+        """Ensure changelog filename extension is used if .md"""
+        setup_simple_project(extra_config='filename = "changes.md"')
+        frag_path = Path("foo", "newsfragments")
+
+        result = runner.invoke(_main, ["123.feature"])
+        self.assertEqual(result.exit_code, 0, result.output)
+
+        fragments = [f.name for f in frag_path.iterdir()]
+        # No '.rst' extension added.
+        self.assertEqual(fragments, ["123.feature.md"])
+
+    @with_isolated_runner
+    def test_odd_filename_extension(self, runner: CliRunner):
+        """Ensure changelog filename extension is not used if not .rst or .md"""
```

Comment:
`Ensure` is kind of redundand here.
We write tests to make sure all works as expected :)

```suggestion
    def test_no_filename_extension(self, runner: CliRunner):
        """
        When the NEWS filename has no extensions,
        new fragments are created by default without any
        extension.
        """
```